### PR TITLE
VACMS-9905: Remove PHPUnit performance tests.

### LINKED ISSUE #9905: Isolate and rework PHPUnit performance tests.
## Description
We have a few PHPUnit performance tests, but they aren't written very well; they just execute some operation and fail if that operation takes longer than a certain amount of time.

This is not great because outlying figures cause test failures that can irritate and distract the developer.

A more statistically defensible test would be to run multiple iterations (say, 10), remove the shortest and longest times, and do a very simple analysis on the remaining times -- mean/median, etc.  It doesn't need to be complex or terribly long; it's just nice to avoid spurious failures like this:

Screen Shot 2022-07-26 at 9 32 09 AM

Another nice thing to do would be to keep these tests in a separate suite.  This would help to minimize the effects of this change on the duration of the CI test runs.

**EDIT**: Tim [suggested](https://dsva.slack.com/archives/CT4GZBM8F/p1666201071140519) that these be a metric with a monitor.  He may be a Xoomer, but he's spitting straight facts here.

## Acceptance Criteria
- [ ] PHPUnit performance tests have been modified as described above.
- [ ] PHPUnit performance tests are moved to a separate suite that is executed concurrently in CI.

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [x] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

### PR DESCRIPTION
**Pull request by ndouglas**
## Description

Closes #9905.

No offense, but these just aren't good tests and I don't believe they're redeemable.

I'll go more into detail on the files page.

My original intent, as reflected in #9905, was to run these tests in a separate suite, sequentially with the other suites rather than concurrently.  But honestly I think they're just bad tests and not terribly meaningful even if I add a `for` loop around them.

If we want performance tests, we should create a story for identifying areas that we want to measure and track over time, we should determine useful ways of measuring them, and then we should write tests that measure them and report the results up to DataDog (or wherever) for tracking.

Currently, these tests are worse than not having the tests at all.  They do not reliably indicate problems, but they semi-reliably cause them.

### CODE REVIEW

File: tests/phpunit/Performance/CreateNodeTest.php [hunk 1]
```diff
@@ -1,61 +0,0 @@
-<?php
-
-namespace tests\phpunit\Performance;
-
-use Tests\Support\Classes\VaGovExistingSiteBase;
```
Comment (ndouglas) [inline tests/phpunit/Performance/CreateNodeTest.php:5]: This test does not do anything useful that isn't done far better and with far greater sensitivity to context in the Cypress tests.  This is not useful.  All it does is cause spurious, random test failures that aggravate engineers and require them to rerun the test until the recalcitrant thing finally passes.

**Review by swirtSJW — COMMENTED** [review]
I am not opposed to removing these and I think your assertions are right about some of the cypress tests giving better coverage.   What do you think about re-purposing these from "performance tests"  to performance **monitoring**.  Where they do the same actions, but just log the time to datadog or something like that.... and only for staging.  So that we can spot trends like  if it takes 45 seconds to login.... or a minute to edit a page.

**Comment by ndouglas** [comment]
@swirtSJW I like the idea of being able to monitor performance trends, but I think it's better done within Cypress than PHPUnit.

My thinking is that we're concerned about two different performance scenarios: editor-facing and content-build-facing.  

I think Tim's adding some good stuff for measuring the content-build metric, but it'd probably be wise to add a content-build-gql elapsed time metric since we're already performing the test, if only so that we can exclude a class of performance issues if we're ever in a blind panic trying to figure out why content builds are taking three hours.  That's probably not that tough.

The editor-facing performance monitoring is best done with Cypress IMHO, but we run into the issue of interference from concurrent tests.  I'm _hoping_ that with #8671 being resolved by #11486 that we'll lose the long tail we've seen in these performance tests on some runs and the numbers will settle down into something approaching a normal distribution.  Admittedly, knowing that loading an edit page _may_ take 30 seconds in some pathological situations is still of some value 🤷🏻‍♂️ 

Anyway, the Cypress tests do involve a whole lot of logging in, loading various node forms, saving nodes, etc, so I think it'd be useful to inject code before and after to measure some of those figures and then post 'em to DataDog.

So, followups:

- #11629 - these accessibility tests don't count login time, just the time to open various pages.  Since each has a unique path, it's trivial to do some text transformation on that and turn it into a metric (e.g. `/node/add/page` becomes `cms_bench._node_add_page` or something).
- #11630 - login time can be measured multiple times over the Cypress test runs.  Because of how long the Cypress tests are, they run concurrently with other tests and also by themselves, so it gives a decent range of conditions.  There might be other steps we can measure -- I'm not thinking of any ATM.
- #11493 - I wanted to ensure behavioral test coverage of creating and saving a node of each content type.  We can't quite get "statistically significant" coverage of this, but the measurements should still be useful, I'd think.
- #11631 - Similar to #11493, but for Media entities.

That's not exactly 1:1 for these performance tests, but I think it'll give better coverage of what matters.  I'm happy to add more issues or extend the ones above if you can think of anything else applicable.  And I think these should be fairly easy to extend in the future.

**Review by swirtSJW — APPROVED** [review]
Thanks for tackling this and spinning up all the new tickets.

**Pull request merged**